tests/unit/live/test_uuid.c

```diff
@@ -67,28 +93,40 @@ TEST(etcpal_uuid, uuid_to_string_conversion_works)
   char str_buf[ETCPAL_UUID_STRING_BYTES];
   EtcPalUuid uuid = {{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16}};
 
-  etcpal_uuid_to_string(str_buf, &uuid);
+  etcpal_uuid_to_string(&uuid, str_buf);
   TEST_ASSERT_EQUAL_STRING(str_buf, "01020304-0506-0708-090a-0b0c0d0e0f10");
 
   uuid = kEtcPalNullUuid;
-  etcpal_uuid_to_string(str_buf, &uuid);
+  etcpal_uuid_to_string(&uuid, str_buf);
   TEST_ASSERT_EQUAL_STRING(str_buf, "00000000-0000-0000-0000-000000000000");
 }
 
 TEST(etcpal_uuid, string_to_uuid_conversion_works)
 {
   EtcPalUuid uuid;
-  const char good_str[] = "08090a0b-0C0D-0e0f-1011-121314151617";
-  const char short_str[] = "08090a0b-0c0d-0e0f-1011-1213141516";
-  const char bad_str[] = "This isn't a UUID";
-
-  TEST_ASSERT(etcpal_string_to_uuid(&uuid, good_str, strlen(good_str)));
-  EtcPalUuid uuid_cmp = {{8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23}};
-  TEST_ASSERT_EQUAL(0, ETCPAL_UUID_CMP(&uuid, &uuid_cmp));
+  const char* good_str = "08090a0b-0C0D-0e0f-1011-121314151617";
+  const EtcPalUuid good_str_uuid = {{8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23}};
+  // clang-format off
+  const char* bad_strings[] = {
+    "08090a0b-0c0d-0e0f-1011-1213141516", // Short by one character
+    "08090a0b-0c0d-0e0f-1011-12131415161g", // Bad character at the end
+    "g8090a0b-0c0d-0e0f-1011-121314151617", // Bad character at the beginning
+    "This isn't a UUID", // Short random string
+    "This isn't a UUID but it is a very long string" // Long random string
+  };
+  // clang-format on
+
+  TEST_ASSERT(etcpal_string_to_uuid(good_str, &uuid));
+  TEST_ASSERT_EQUAL(0, ETCPAL_UUID_CMP(&uuid, &good_str_uuid));
 
   uuid = kEtcPalNullUuid;
-  TEST_ASSERT_UNLESS(etcpal_string_to_uuid(&uuid, short_str, strlen(short_str)));
-  TEST_ASSERT_UNLESS(etcpal_string_to_uuid(&uuid, bad_str, strlen(bad_str)));
+
+  for (size_t i = 0; i < (sizeof(bad_strings) / sizeof(const char*)); ++i)
```

Comment:
You can do ranged for on bad_strings. std::size(bad_strings) will also get you the size, rather than having to do the sizeof trick.

Reply:
Not in a C file, you can't!